Jenkinsfile

```diff
@@ -48,129 +44,103 @@ pipeline {
                     expression { env.CHANGE_TARGET != null }
                 }
             }
-            // TODO DDF-2971 refactor this stage from scripted syntax to declarative syntax to match the rest of the stages - https://issues.jenkins-ci.org/browse/JENKINS-41334
-            steps {
-                parallel(
-                    linux: {
-                        node('linux-large') {
-                            retry(3) {
-                                checkout scm
-                            }
-                            timeout(time: 3, unit: 'HOURS') {
-                                // TODO: Maven downgraded to work around a linux build issue. Falling back to system java to work around a linux build issue. re-investigate upgrading later
-                                withMaven(maven: 'Maven 3.3.9', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LARGE_MVN_OPTS} ${LINUX_MVN_RANDOM}', options: [artifactsPublisher(disabled: true), dependenciesFingerprintPublisher(disabled: true, includeScopeCompile: false, includeScopeProvided: false, includeScopeRuntime: false, includeSnapshotVersions: false)]) {
-                                    sh 'mvn install -B -pl !$DOCS -DskipStatic=true -DskipTests=true -T 1C'
-                                    sh 'mvn clean install -B -T 1C -pl !$ITESTS -Dgib.enabled=true -Dgib.referenceBranch=/refs/remotes/origin/$CHANGE_TARGET'
-                                    sh 'mvn install -B -pl $ITESTS -nsu'
-                                }
+            parallel {
```

Comment:
@mcalcote another thing we might want to consider, now that the parallel steps actually fit the declarative syntax, is that we could actually collapse the `Incremental Build` and `Full Build` Stages into a single stage with multiple conditional parallel branches
Since each parallel stage supports the full syntax we could move the conditions from the parent stage into each parallel stage making the pipeline a little simpler. We would then have a single build stage that would run different parallel stages depending on whether it was a PR or a regular branch

```
stage('Build') {
  parallel {
    stage('Linux Full Build') {
      when { expression { env.CHANGE_ID == null } }
      ...
    }
    stage('Windows Full Build') {
      when { expression { env.CHANGE_ID == null } }
      ...
    }
    stage('Linux Incremental Build') {
      when {
        allOf {
          expression { env.CHANGE_ID != null }
          expression { env.CHANGE_TARGET != null }
        }
        ...
      }
      stage('Windows Incremental Build') {
      when {
        allOf {
          expression { env.CHANGE_ID != null }
          expression { env.CHANGE_TARGET != null }
        }
        ...
      }
    }
  }
}
```

Reply:
One downside of this is that given the current way jenkins reports build progress across nodes it would appear that all 4 stages are being run. In the  current arrangement it at least shows half of the nodes being skipped. I can change it to this, but I want to make sure we are alright with that.